### PR TITLE
Fix xml reports changed when node is not deleted

### PR DESCRIPTION
**Pull request by mklassen**
##### SUMMARY
`xml` module reports changed even if the xpath to be deleted already does not exist.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
xml

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Change looks good (without knowing the internals of the `xml` module :) ). You need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to), and it would be great if you could add a test for this (the tests for the module are in https://github.com/ansible-collections/community.general/tree/main/tests/integration/targets/xml).

**Comment by mklassen** [comment]
Changelog fragment and integration tests have been added.

File: changelogs/fragments/xml-remove-changed.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - xml - Fixed issue were changed was returned when removing non-existent xpath
```
Comment (felixfontein) [inline changelogs/fragments/xml-remove-changed.yml:2]:
```suggestion
  - xml - fixed issue were changed was returned when removing non-existent xpath (https://github.com/ansible-collections/community.general/pull/1007).
```

File: tests/integration/targets/xml/tasks/test-remove-attribute-nochange.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      dest: /tmp/ansible-xml-beers.xml
+    check_mode: yes
+    diff: yes
+    register: comparison
+
+  - name: Test expected result
+    assert:
+      that:
+      - remove_attribute.changed == false
+      - comparison.changed == false  # identical
```
Comment (felixfontein) [inline tests/integration/targets/xml/tasks/test-remove-attribute-nochange.yml:27]: The standard Ansible test way would be:
```suggestion
      - remove_attribute is not changed
      - comparison is not changed  # identical
```
What you did is consistent with what the other tests for `xml` do though. Feel free to ignore this comment.

Comment (mklassen) [reply tests/integration/targets/xml/tasks/test-remove-attribute-nochange.yml:27]: Yes, I choose consistency over modernization.

**Review by felixfontein — APPROVED** [review]

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@mklassen thanks for the fix!
@felixfontein thanks for reviewing!

**Pull request merged**